app/src/main/java/com/example/todolist/utils/SynchronousExecutorService.java

```diff
@@ -0,0 +1,41 @@
+package com.example.todolist.utils;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.AbstractExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class SynchronousExecutorService extends AbstractExecutorService {
+    private volatile boolean terminated;
+
+    @Override
+    public void shutdown() {
+        terminated = true;
+    }
+
+    @Override
+    public boolean isShutdown() {
+        return terminated;
+    }
+
+    @Override
+    public boolean isTerminated() {
+        return terminated;
+    }
+
+    @Override
+    public boolean awaitTermination(long theTimeout, TimeUnit theUnit) {
+        shutdown();
+        return terminated;
+    }
+
+    @Override
+    public List<Runnable> shutdownNow() {
+        return Collections.emptyList();
+    }
+
+    @Override
+    public void execute(Runnable theCommand) {
+        theCommand.run();
+    }
```

Comment:
Ты можешь это упростить
```suggestion
    @Override
    public void shutdown() { }

    @Override
    public boolean isShutdown() {
        return false;
    }

    @Override
    public boolean isTerminated() {
        return false;
    }

    @Override
    public boolean awaitTermination(long theTimeout, TimeUnit theUnit) {
        return true;
    }

    @Override
    public List<Runnable> shutdownNow() {
        return Collections.emptyList();
    }

    @Override
    public void execute(Runnable theCommand) {
        theCommand.run();
    }
```